extension/src/init.js

```diff
@@ -1,22 +1,28 @@
+const pMap = require('p-map')
+
 const server = require('./server.js').setupServer()
 const utils = require('./utils')
-const configLoader = require('./config/config')
+const config = require('./config/config')
 const ctpClientBuilder = require('./ctp')
 
-const config = configLoader.load()
+const envConfig = config.getEnvConfig()
 
 const { ensureResources } = require('./config/init/ensure-resources')
 
-const port = parseInt(config.port || 8080, 10)
+const port = parseInt(envConfig.port || 8080, 10)
 const logger = utils.getLogger()
 
-// raise an exception when there are no CTP credentials
-if (!config.ctp.projectKey || !config.ctp.clientId || !config.ctp.clientSecret)
-  throw new Error('CTP project credentials are missing')
-
-if (config.keepAliveTimeout !== undefined)
-  server.keepAliveTimeout = config.keepAliveTimeout
+if (envConfig.keepAliveTimeout !== undefined)
+  server.keepAliveTimeout = envConfig.keepAliveTimeout
 server.listen(port, async () => {
-  await ensureResources(ctpClientBuilder.get())
+  const ctpProjectKeys = config.getAllCtpProjectKeys()
+  await pMap(ctpProjectKeys, async (ctpProjectKey) => {
+    const ctpConfig = config.getCtpConfig(ctpProjectKey)
+    if (ctpConfig.ensureResources) {
+      ctpClientBuilder.get(ctpProjectKey)
+      await ensureResources()
```

Comment:
could you put todos in code if it's not done yet ? 🤔 

```suggestion
      await ensureResources(ctpClient, url)
```

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/526/commits/65b4483aec8131465a62654906d3457e95ab4c85